src/web/tools/KeyEventManager.ts

```diff
@@ -0,0 +1,103 @@
+import { AdaptedEvent, EventTypes } from '../interfaces';
+import EventManager from './EventManager';
+import { PointerType } from '../../PointerType';
+import { View } from 'react-native';
+
+export default class KeyEventManager extends EventManager<HTMLElement> {
+  private activationKeys = ['Enter', 'Space', 'NumpadEnter'];
+  private cancelationKeys = ['Tab'];
+
+  private keyDownCallback = (event: KeyboardEvent): void => {
+    console.log(event.code);
+    if (this.cancelationKeys.indexOf(event.code) !== -1) {
+      this.dispatchEvent(event, EventTypes.CANCEL);
```

Comment:
Have you tested a scenario where one view is focused, Enter is pressed down, tab is pressed to focus another view, and then Enter is lifted?

If I understand correctly, the first view would receive DOWN and CANCEL events, but the second one would only receive an UP event. If that's the case, have you tested how different handlers behave in that scenario?

Reply:
This cancelation logic handles the case you've mentioned. 
Without it, it was possible to press `Enter`, switch focus with `Tab` and after that it was impossible to use `Enter` again, and the previously pressed element stayed in a `pressed down` state.

As for an element only receiving the `UP` event, I haven't observed any concerning behaviour.
Afaik both `Gesture Handler` and native buttons handle such cases, but i'll fix this just to be safe.

Reply:
As of [044cf87](https://github.com/software-mansion/react-native-gesture-handler/pull/3035/commits/044cf87ff6b36207e1794f07d731f6d0f1b4724a) added logic preventing calling UP and CANCEL when element is not pressed down.